src/plugins/shared_ux/public/components/page_template/no_data_page/no_data_card/elastic_agent_card.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { ElasticAgentCardProps } from './types';
+import { useHttp, usePermissions } from '../../../../services';
+import { ElasticAgentCardComponent } from './elastic_agent_card.component';
+
+export const ElasticAgentCard = (props: ElasticAgentCardProps) => {
+  const { canAccessFleet } = usePermissions();
+  const { addBasePath } = useHttp();
+
+  const createHref = () => {
+    const { href, category } = props;
+    if (href) {
+      return href;
+    }
+    const prefix = '/app/integrations/browse';
```

Comment:
is there a way to make this not-hardcoded in the component?

Reply:
All of this could be mitigated if integrations provided a Locator: https://github.com/elastic/kibana/blob/main/src/plugins/share/common/url_service/locators/README.md

It looks like a lot of people hardcode this.  It would be great if there was a locator.  Might want to reference a TODO here.

cc: @elastic/fleet

Reply:
We are tracking this here, but it hasn't been a high-priority item. It'd be helpful to understand what is at stake by not using these on the issue" https://github.com/elastic/kibana/issues/102740